src/commands/Minion/alch.ts

```diff
@@ -0,0 +1,146 @@
+import { CommandStore, KlasaMessage } from 'klasa';
+import { Util } from 'oldschooljs';
+import { Item } from 'oldschooljs/dist/meta/types';
+
+import { BotCommand } from '../../lib/BotCommand';
+import { UserSettings } from '../../lib/settings/types/UserSettings';
+import resolveItems from '../../lib/util/resolveItems';
+import hasItemEquipped from '../../lib/gear/functions/hasItemEquipped';
+import {
+	bankHasAllItemsFromBank,
+	formatDuration,
+	removeBankFromBank,
+	resolveNameBank
+} from '../../lib/util';
+import createReadableItemListFromBank from '../../lib/util/createReadableItemListFromTuple';
+import addSubTaskToActivityTask from '../../lib/util/addSubTaskToActivityTask';
+import { AlchingActivityTaskOptions } from '../../lib/types/minions';
+import { Activity, Tasks, Time } from '../../lib/constants';
+import { rand } from '../../util';
+
+const options = {
+	max: 1,
+	time: 10_000,
+	errors: ['time']
+};
+
+const unlimitedFireRuneProviders = resolveItems([
+	'Staff of fire',
+	'Fire battlestaff',
+	'Mystic fire staff',
+	'Lava battlestaff',
+	'Mystic lava staff',
+	'Steam battlestaff',
+	'Mystic steam staff',
+	'Smoke battlestaff',
+	'Mystic smoke staff',
+	'Tome of fire'
+]);
+
+export default class extends BotCommand {
+	public constructor(store: CommandStore, file: string[], directory: string) {
+		super(store, file, directory, {
+			cooldown: 1,
+			usage: '[quantity:int{1}] (item:...item)',
+			usageDelim: ' '
+		});
+	}
+
+	async run(msg: KlasaMessage, [quantity = null, item]: [number | null, Item[]]) {
+		const userBank = msg.author.settings.get(UserSettings.Bank);
+		const osItem = item.find(i => userBank[i.id] && i.highalch);
+		if (!osItem) {
+			throw `'You don't have any of this item to alch.'`;
+		}
+		// 5 tick action
+		const timePerAlch = Time.Second * 3;
+
+		if (!quantity) {
```

Comment:
move this below the maxCasts calc and set quantity to maxCasts, otherwise this could automatically set quant to something greater than you can do